logging/impl.go

```diff
@@ -198,6 +223,42 @@ func (imp *impl) shouldLog(logLevel Level) bool {
 }
 
 func (imp *impl) Write(entry *LogEntry) {
+	if imp.dedupNoisyLogs {
+		// If we have have entered a new recentMessage window, output noisy logs from
+		// the last window.
+		if time.Since(imp.recentMessageWindowStart) > noisyMessageWindowDuration {
```

Comment:
I know this is not a PR, but just noting that I think most of this code would need to be wrapped in a lock.

Reply:
Lock added.